### PR TITLE
Homework2: Natalia Mashliakevich

### PR DESCRIPTION
**Pull request by nataliasipenko**
- [x] ### PR checklist

Please make sure you've passed PR checklist
- [x] my PR passes CI checks (flake8 and pycodestyle).
- [x] my PR named according to guidelines.
- [x] my PR contains complete tasks descriptions for all tasks.
- [x] my PR made from the fork feature branch to appropriate branch.
- [x] my PR contains only code that solves appropriate homework tasks. Redundant 
files (configs, etc. ) are put to .gitignore.

### CODE REVIEW

File: src/homework2/task1.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+M = int(input('Введите количество рублей (M): '))
+N = int(input('Введите количество копеек (N): '))
+S = int(input('Введите количество товара (S): '))
+
+print('Общая цена за', S, 'шт. составляет', M, 'рублей(-я)', N, 'копеек')
+
+price_for_one_item_cents = ((M * 100) + N) / S
+
+# Подсчет общей цены в рублях и копейках за L товаров
```
Comment (belovmd) [inline src/homework2/task1.py]: redundant comments

File: src/homework2/task1.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+Напишите программу, которая считает общую цену.
+Вводится M рублей и N копеек цена, а также количество S товара.
+Посчитайте общую цену в рублях и копейках за L товаров.
+"""
+
+M = int(input('Введите количество рублей (M): '))
+N = int(input('Введите количество копеек (N): '))
+S = int(input('Введите количество товара (S): '))
+
+print('Общая цена за', S, 'шт. составляет', M, 'рублей(-я)', N, 'копеек')
```
Comment (belovmd) [inline src/homework2/task1.py:11]: i don't understand this pring

File: src/homework2/task2.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+"""
+Найти самое длинное слово в введенном предложении.
+Учтите что в предложении есть знаки препинания.
+"""
+
+sentence = str(input('Введите предложение со знаками препинания:'))
+
+punctuation = ['(', ')', '?', ':', ';', ',', '.', '!', '/', '"', "'"]
```
Comment (belovmd) [inline src/homework2/task2.py]: what about > { and many others?

File: src/homework2/task5.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+"""
+
+fibonacci_num_1 = 1
+fibonacci_num_2 = 1
+
+fibonacci_n = int(input('Введите n (номер элемента ряда Фибоначчи):'))
+
+i = 2
+while i < fibonacci_n:
+    fibonacci_sum = fibonacci_num_1 + fibonacci_num_2
```
Comment (belovmd) [inline src/homework2/task5.py]: a, b = a+b, a

File: src/homework2/task6.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+Число положительное целое, произвольной длины. Задача требует работать только с числами
+(без конвертации числа в строку или что-нибудь еще)
+"""
+
+number = int(input('Введите целое положительное число произвольной длины: '))
+reversion_of_number = 0
+n = number
+
+if number < 10:
+    print('С однозначным числом операция не выполняется')
```
Comment (belovmd) [inline src/homework2/task6.py]: all 1-digit numbers are palindrom

File: src/homework2/task7.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+найти его площадь. Если нет, вывести сообщение о неверных данных.
+"""
+import math
+
+a = float(input('Введите одну сторону треугольника: '))
+b = float(input('Введите вторую сторону треугольника: '))
+c = float(input('Введите третью сторону треугольника: '))
+
+if a + b > c and b + c > a and c + a > b:
+    perimeter = float((a + b + c) / 2)
```
Comment (belovmd) [inline src/homework2/task7.py]: it's semi-perimeter

Comment (belovmd) [reply src/homework2/task7.py]: redundant `float()`

File: src/homework2/task8.py [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+Otherwise return NO.
+"""
+
+
+def tickets(people):
+    d25 = 0
+    d50 = 0
+    d100 = 0
+    for every in people:
+        if every == 25:
```
Comment (belovmd) [inline src/homework2/task8.py]: person in people

File: src/homework2/task8.py [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+            d25 += 1
+        elif every == 50:
+            if d25 >= 1:
```
Comment (belovmd) [inline src/homework2/task8.py]: if d25?

File: src/homework2/task8.py [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+                d50 += 1
+                d25 -= 1
+            else:
+                return "NO"
+        else:
+            if d50 >= 1 and d25 >= 1:
```
Comment (belovmd) [inline src/homework2/task8.py]: if d50?

File: src/homework2/task8.py [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+create_phone_number([1, 2, 3, 4, 5, 6, 7, 8, 9, 0]) # => returns "(123) 456-7890"
+
+The returned format must be correct in order to complete this challenge.
+Don't forget the space after the closing parentheses!
+"""
+
+
+def create_phone_number(n):
+    template = "({}{}{}) {}{}{}-{}{}{}{}"
+    return template.format(*n)
```
Comment (belovmd) [inline src/homework2/task8.py:82]: nice

File: src/homework2/task8.py [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+def digital_root(n):
+    summa = 0
+
+    if n > 9:
+        while n > 0:
+            digit = n % 10
+            summa += digit
+            n = n // 10
+
+            if summa > 9:
```
Comment (belovmd) [inline src/homework2/task8.py]: same code as above. why couldn't we do it in one while?

File: src/homework2/task8.py [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+ of 3 or 5 below the number passed in.
+Note: If the number is a multiple of both 3 and 5, only count it once.
+Also, if a number is negative, return 0(for languages that do have them)
+
+"""
+
+
+def solution(number):
+    if number < 0:
+        print(0)
```
Comment (belovmd) [inline src/homework2/task8.py]: return 0

File: src/homework2/task8.py [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+    else:
+        summa = 0
+        for element in range(number):
+            if element % 3 == 0 or element % 5 == 0:
+                summa += element
+        return summa
+
+# Короткое решение:
```
Comment (belovmd) [inline src/homework2/task8.py]: please chose one solution

**Review by belovmd — CHANGES_REQUESTED** [review]
- Sanity checks failed - redundant files in PR
- it would be nice to have one 5 kyu task for higher score
- please address next issues:

File: src/homework2/task1.py (hunk 2 above)
Comment (nataliasipenko) [reply src/homework2/task1.py:11]: _Напишите программу, которая считает общую цену. Вводится **M** рублей и **N** копеек цена, а также количество **S** товара. Посчитайте общую цену в рублях и копейках за **L** товаров._ We have some amount of goods (S) and final price of this quantity.  Firstly we should find price for one item (price for all items ((m*100) + n) / s). Then we get another quantity of goods (L) and should find out, how much is this quantity (L) of goods.

**Review by nataliasipenko — COMMENTED** [review]
Changes made. Review requested (please see my comment below).

File: src/homework2/task1.py (hunk 2 above)
Comment (belovmd) [reply src/homework2/task1.py:11]: yes but this print goes before calculation

File: src/homework2/task7.py [hunk 14]
```diff
@@ -0,0 +1,17 @@
...
+найти его площадь. Если нет, вывести сообщение о неверных данных.
+"""
+import math
+
+a = float(input('Введите одну сторону треугольника: '))
+b = float(input('Введите вторую сторону треугольника: '))
+c = float(input('Введите третью сторону треугольника: '))
+
+if a + b > c and b + c > a and c + a > b:
+    perimeter = (a + b + c)
```
Comment (belovmd) [inline src/homework2/task7.py:13]: it should be a semi-perimeter. 
it gives wrong answer

**Review by belovmd — APPROVED** [review]
8,5

**Pull request closed without merging**